src/utils/osx/OSXUtils.mm

```diff
@@ -81,3 +82,22 @@
   else
     [NSCursor hide];
 }
+
+/////////////////////////////////////////////////////////////////////////////////////////
+bool OSXUtils::isWindowFullScreen(KonvergoWindow *qtWindow)
+{
+  NSView* cocoaView = (NSView*)qtWindow->winId();
+  NSWindow* cocoaWindow = cocoaView.window;
+  return cocoaWindow.styleMask & NSFullScreenWindowMask;
+}
+
+/////////////////////////////////////////////////////////////////////////////////////////
+void OSXUtils::SetWindowFullScreen(KonvergoWindow *qtWindow, bool fullScreen)
+{
+  if (fullScreen == isWindowFullScreen(qtWindow))
+    return;
+
+  NSView* cocoaView = (NSView*)qtWindow->winId();
```

Comment:
nullcheck for window
you also most likely want to make sure  you get a cocoaView valid pointer here ?